packages/flutter/lib/src/material/time_picker_theme.dart

```diff
@@ -0,0 +1,273 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines the visual properties of the widget displayed with [showTimePicker].
+///
+/// Descendant widgets obtain the current [TimePickerThemeData] object using
+/// `TimePickerTheme.of(context)`. Instances of [TimePickerThemeData]
+/// can be customized with [TimePickerThemeData.copyWith].
+///
+/// Typically a [TimePickerThemeData] is specified as part of the overall
+/// [Theme] with [ThemeData.timePickerTheme].
+///
+/// All [TimePickerThemeData] properties are `null` by default. When null,
+/// [showTimePicker] will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class TimePickerThemeData with Diagnosticable {
+
+  /// Creates a theme that can be used for [TimePickerTheme] or
+  /// [ThemeData.timePickerTheme].
+  const TimePickerThemeData({
+    this.backgroundColor,
+    this.headerColor,
+    this.dialHandColor,
+    this.dialBackgroundColor,
+    this.activeDayPeriodColor,
+    this.hourMinuteTextStyle,
+    this.dayPeriodTextStyle,
+    this.helpTextStyle,
+    this.shape,
+    this.hourMinuteShape,
+    this.dayPeriodShape,
+    this.inputDecorationTheme,
+  });
+
+  /// The background color of a time picker.
+  ///
+  /// If this is null, the time picker defaults to [ColorScheme.background].
+  final Color backgroundColor;
+
+  /// The color used in the header of a time picker.
+  ///
+  /// This determines the active color of the header segments that represent
+  /// hours and minutes.
+  ///
+  /// If this is null, the time picker defaults to [ColorScheme.primary].
+  final Color headerColor;
+
+  /// The color of the time picker dial's hand.
+  ///
+  /// If this is null, the time picker defaults to [ColorScheme.primary].
+  final Color dialHandColor;
+
+  /// The background color of the time picker dial.
+  ///
+  /// If this is null, the time picker defaults to [ColorScheme.primary].
+  final Color dialBackgroundColor;
+
+  /// The background color of the active day period in the time picker.
+  ///
+  /// If this is null, the time picker defaults to [ColorScheme.surface].
+  final Color activeDayPeriodColor;
+
+  /// Used to configure the [TextStyle]s for the hour/minute controls.
+  ///
+  /// If this is null, the time picker defaults to
+  /// `Theme.of(context).textTheme.headline3`.
```

Comment:
Is it common to say Theme.of(context).textTheme.foo here or just TextTheme.foo?

Reply:
you're right, looks like [ThemeData.textTheme.foo] is the pattern